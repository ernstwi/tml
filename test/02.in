foo
init bar invar { x <= 10 }

foo -> bar
    guard { x - y == 5 }
    sync { press }
bar -> baz
    sync { press }
baz -> baz
