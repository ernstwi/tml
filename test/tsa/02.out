{
    "edges": [
        {
            "from": "bar",
            "guard": null,
            "reset": null,
            "sync": {
                "id": "press",
                "type": "internal"
            },
            "to": "baz"
        },
        {
            "from": "baz",
            "guard": null,
            "reset": null,
            "sync": null,
            "to": "baz"
        },
        {
            "from": "foo",
            "guard": "x-y==5",
            "reset": null,
            "sync": {
                "id": "press",
                "type": "internal"
            },
            "to": "bar"
        }
    ],
    "locations": [
        {
            "id": "bar",
            "initial": true,
            "invariant": "x<=10"
        },
        {
            "id": "foo",
            "initial": false,
            "invariant": null
        }
    ]
}
